packages/react-router/modules/matchPath.js

```diff
@@ -1,5 +1,59 @@
 import pathToRegexp from "path-to-regexp";
 
+function isAbsolute(pathname) {
+  return pathname.charAt(0) === "/";
+}
+
+function ensureTrailingSlash(pathname) {
+  return hasTrailingSlash(pathname) ? pathname : pathname + "/";
+}
+
+function hasTrailingSlash(pathname) {
+  return pathname.charAt(pathname.length - 1) === "/";
+}
+
+function isMalformed(pathname) {
```

Comment:
isMalFormed and refersToParentSegment are theoretical redundant to each other.

just checking `pathname.slice(0,2) === ".."` is enough to identify an illegal path. 

Depends on whether or not we really want to distinguish the two cases. I think that telling the user that paths can't start with two dots is probably enough.

Reply:
Right, we could shrink this to one case.